File: web/src/types/attachment.ts
/**
 * Attachment metadata — mirrors Android Attachment model.
 * Binary payload lives in object storage (`storagePath`: `r2:` or `pending:` prefix on Web).
 */
export interface Attachment {
  id: string;
  noteId: number;
  storagePath: string;
  type: string;
  mimeType?: string;
  sizeBytes?: number;
}

export interface AttachmentCloudMap {
  storagePath: string;
  type: string;
  mimeType?: string;
  sizeBytes?: number;
}

File: composeApp/src/commonMain/kotlin/com/aus/notelikeus/domain/model/Attachment.kt
package com.aus.notelikeus.domain.model

import androidx.compose.runtime.Immutable
import kotlinx.serialization.Serializable

/**
 * Attachment metadata. Binary payload lives in local files or object storage
 * ([storagePath]: `pending:`, `r2:`, or `file:` prefix).
 */
@Immutable
@Serializable
data class Attachment(
    val id: String,
    val noteId: Long,
    val storagePath: String,
    val type: String = "image",
    val mimeType: String? = null,
    val sizeBytes: Long? = null,
)
